sdk/textanalytics/ai-text-analytics/swagger/README.md

```diff
@@ -550,3 +472,120 @@ directive:
     transform: $.description = "An object that describes metadata about the healthcare entity such as whether it is hypothetical or conditional.";
 ```
+
+## JS customizations
+
+### Task renames
+
+```yaml
+directive:
+  - from: swagger-document
+    where: $.definitions.AnalyzeTextTask
+    transform:
+      $["x-ms-client-name"] = "AnalyzeAction";
+  - from: swagger-document
+    where: $.definitions.AnalyzeTextLROTask
+    transform:
+      $["x-ms-client-name"] = "AnalyzeBatchAction";
+  - from: swagger-document
+    where: $.definitions.EntityLinkingTaskParameters
+    transform:
+      $["x-ms-client-name"] = "EntityLinkingAction";
+  - from: swagger-document
+    where: $.definitions.EntitiesTaskParameters
+    transform:
+      $["x-ms-client-name"] = "EntityRecognitionAction";
+  - from: swagger-document
+    where: $.definitions.SentimentAnalysisTaskParameters
+    transform:
+      $["x-ms-client-name"] = "SentimentAnalysisAction";
+  - from: swagger-document
+    where: $.definitions.KeyPhraseTaskParameters
+    transform:
+      $["x-ms-client-name"] = "KeyPhraseExtractionAction";
+  - from: swagger-document
+    where: $.definitions.PiiTaskParameters
+    transform:
+      $["x-ms-client-name"] = "PiiEntityRecognitionAction";
+  - from: swagger-document
+    where: $.definitions.LanguageDetectionTaskParameters
+    transform:
+      $["x-ms-client-name"] = "LanguageDetectionAction";
+  - from: swagger-document
+    where: $.definitions.HealthcareTaskParameters
+    transform:
+      $["x-ms-client-name"] = "HealthcareAction";
+  - from: swagger-document
+    where: $.definitions.ExtractiveSummarizationTaskParameters
+    transform:
+      $["x-ms-client-name"] = "ExtractiveSummarizationAction";
+  - from: swagger-document
+    where: $.definitions.ExtractiveSummarizationSortingCriteria
+    transform:
+      $["x-ms-enum"].name = "ExtractiveSummarizationOrderingCriteria";
+  - from: swagger-document
+    where: $.definitions.CustomEntitiesTaskParameters
+    transform:
+      $["x-ms-client-name"] = "CustomEntityRecognitionAction";
+  - from: swagger-document
+    where: $.definitions.CustomSingleLabelClassificationTaskParameters
+    transform:
+      $["x-ms-client-name"] = "CustomSingleLabelClassificationAction";
+  - from: swagger-document
+    where: $.definitions.CustomMultiLabelClassificationTaskParameters
+    transform:
+      $["x-ms-client-name"] = "CustomMultiLabelClassificationAction";
+  - from: swagger-document
+    where: $.definitions.TaskIdentifier
+    transform:
+      $["x-ms-client-name"] = "BatchActionState";
+  - from: swagger-document
+    where: $.definitions..properties.stringIndexType
+    transform: >
+      $.description = "Specifies the measurement unit used to calculate the offset and length properties. For a list of possible values, see {@link KnownStringIndexType}.\n\nThe default is the JavaScript's default which is \"Utf16CodeUnit\".";
+```
+
+### Rename JobState
+
+```yaml
+directive:
```

Comment:
In other SDKs I've worked on, these were definitely cross-language. Shouldn't they stay in `General`?

Also, in the past we've simply used `modifiedOn` not `lastModifiedOn`, since "last" is redundant anyway.

Reply:
I am not sure tbh but I am happy to move those up if all other languages follow those names. /cc @mssfang @kristapratico

Reply:
> Also, in the past we've simply used modifiedOn not lastModifiedOn, since "last" is redundant anyway.

Good point, I am addressing this in https://github.com/Azure/azure-sdk-for-js/pull/22311

Reply:
@kristapratico Thanks for confirming. How about the other names (e.g. createdOn, expiresOn, etc), does Python use those?

Reply:
We use `createdOn` and `expiresOn`, but call `operationId` just `id`. We don't expose `status` since it's a method on our poller.

Reply:
In Java, we GAed with ...At. You can see it from API view here: https://apiview.dev/Assemblies/Review/fe6cd393a9bb4a869c220e353b41d4e4/018667882a544f62bca8aad30d640dd7?diffRevisionId=e128928e06284db585cc3aecadc53a57&doc=False&diffOnly=False#com.azure.ai.textanalytics.models.AnalyzeActionsOperationDetail

Reply:
I think we used to discuss the naming of it and then decided to stay with what makes sense to the programming language.

Reply:
Thanks folks, seems like the discrepancies between languages on this justifies keeping this transform under the JS-specific section.

Reply:
This also seems to support the idea I've raised with @tg-msft, @johanste, et. al. about keeping some data base of common patterns. For Key Vault - one of the first track 2s we shipped as a team - architects discussed different ideas for things and wanted us to be consistent across languages. But seems after a time, they forget that.

Ted, Johan, what about a separate page/section of our existing guidelines for common word transforms like these?

(Just to clarify, I agree: keep this in the JS section for now.)

Reply:
I'd love to put this in the REST style guide so it's seen early enough for service teams to change the name when possible.  We've already got https://github.com/microsoft/api-guidelines/issues/368 filed to track that.

Reply:
Also, another idea: if there is some standard REST name that the architects agreed on naming differently in the SDK side, we could bake a transform for it in code gen tooling.

Reply:
I added a comment to track a few I remember being discussed at length: https://github.com/microsoft/api-guidelines/issues/368#issuecomment-1164970962